core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -256,9 +292,46 @@ public void initialize(String name, Map<String, String> unresolved) {
             mergedProps,
             RESTCatalogProperties.METRICS_REPORTING_ENABLED,
             RESTCatalogProperties.METRICS_REPORTING_ENABLED_DEFAULT);
+
+    this.tableCache = tableCacheBuilder(mergedProps).build();
+
     super.initialize(name, mergedProps);
   }
 
+  @VisibleForTesting
+  Caffeine<Object, Object> tableCacheBuilder(Map<String, String> props) {
```

Comment:
nit: Why is this Object, Object?

Reply:
I just grepped for `Caffeine<` and apparently <Object, Object> is how we create these builders. I can change this, but I'd rather follow the pattern.